zcncore/transactionauth_base.go

```diff
@@ -32,7 +32,7 @@ func newTransactionWithAuth(cb TransactionCallback, txnFee uint64, nonce int64)
 }
 
 func (ta *TransactionWithAuth) getAuthorize() (*transaction.Transaction, error) {
-	ta.t.txn.PublicKey = _config.wallet.Keys[0].PublicKey
+	ta.t.txn.PublicKey = _config.wallet.ClientKey
```

Comment:
The `sign()` method below will use `_config.wallet.Key[0].PrivateKey` to sign the transaction, so if it is possible that the `ClientKey` is not the `Keys[0].PublicKey`, then we will use the wrong key to sign the transaction. My question is: is it possible the `_config.wallet.Keys` be in different order that the first key is not the `ClientKey`? If it's always the `ClientKey`, then we actually don't need this changes, otherwise, we'd better make sure the order is guaranteed.

Reply:
What's the relationship between the primary public key and the bolt's public. All keys in the wallet should be derived from the same seed. Otherwise we need to separate the keys.

Reply:
@peterlimg check detail from https://github.com/0chain/gosdk/blob/7140de11080e5b2e45fd6545a0fb662b77fd713f/core/zcncrypto/bls0chain_herumi_test.go#L114

Reply:
Keys are spilt from primary private key

Reply:
Please also check setupAuth method for more context

Reply:
Oh no, who gave the instructions about the implementation of the `SplitKeys`? The `SplitKey` is not deterministic, which means gives the same primary key, running it multiple times would give random keys. This will make the wallet unable to recovery from seed, and users can lost their tokens if they think they can recover all keys from seed.

Reply:
on bolt, it will ask to recover primary key, and split new sub keys for 2FA. recover sub keys seems not be required there. @guruhubb please confirm it.thanks.